config/configload/merge.go

```diff
@@ -592,6 +621,34 @@ func absInBackends(block *hclsyntax.Block) error {
 	return nil
 }
 
+func addProxy(block *hclsyntax.Block, proxies map[string]*hclsyntax.Block) error {
+	if attr, ok := block.Body.Attributes[proxy]; ok {
+		reference, err := attrStringValue(attr)
+		if err != nil {
+			return err
+		}
+
+		delete(block.Body.Attributes, proxy)
+
+		if proxyBlock, ok := proxies[reference]; !ok {
+			return newMergeError(errMissingReferencedProxy, block)
+		} else {
+			block.Body.Blocks = append(block.Body.Blocks, proxyBlock)
+		}
+	}
+
+	return nil
+}
+
+func attrStringValue(attr *hclsyntax.Attribute) (string, error) {
+	v, err := eval.Value(nil, attr.Expr)
+	if err != nil {
+		return "", err
+	}
+
+	return v.AsString(), nil
```

Comment:
```
proxy = 1
```
Check for `cty.String` type before using `.AsString()`

Reply:
Done